### DISCUSSION TITLE
The box "recurring" is already open and of type Box<Goal Model>.

### DISCUSSION THREAD

**Discussion by Madhan790** [community/community#121206] [Programming Help]
### Select Topic Area

Question

### Body

E/flutter (32529): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: HiveError: The box "recurring" is already open and of type Box<Goal
Model>.
E/flutter (32529): #0      HiveImpl._getBoxInternal (package:hive/src/hive_impl.dart:182:9) 
E/flutter (32529): #1      HiveImpl.box (package:hive/src/hive_impl.dart:197:33)
E/flutter (32529): #2      HiveImpl._openBox (package:hive/src/hive_impl.dart:80:16) 
E/flutter (32529): #3      HiveImpl.openBox (package:hive/src/hive_impl.dart:142:18)
E/flutter (32529): #4      HiveBoxModule.recurringBox (package:paisa/di/module/hive_module.dart:49:12)
E/flutter (32529): #5      init.<anonymous closure> (package:paisa/di/dependency_injection.config.dart:206:25)
E/flutter (32529): #6      GetItHelper.lazySingletonAsync (package:injectable/src/get_it_helper.dart:166:27)
E/flutter (32529): #7      init (package:paisa/di/dependency_injection.config.dart:205:12)
E/flutter (32529): <asynchronous suspension>
E/flutter (32529): #8      main (package:paisa/main.dart:27:3)
E/flutter (32529): <asynchronous suspension>
E/flutter (32529):

The error message indicates that there is an issue with Hive, a popular NoSQL database for Flutter applications. Specifically, it's trying to open a box named "recurring" which is already open and of type Box<GoalModel>. This suggests a problem with the initialization or management of Hive boxes in my app.

**Comment by Madhan790** [comment]
// Dart imports:
import 'dart:math';

// Flutter imports:
import 'package:flutter/foundation.dart';
import 'package:flutter/material.dart';

// Package imports:
import 'package:get_it/get_it.dart';

// Project imports:
import 'package:paisa/app.dart';
import 'package:paisa/core/common_enum.dart';
import 'package:paisa/di/dependency_injection.dart';
import 'package:paisa/features/account/data/data_sources/account_data_source.dart';
import 'package:paisa/features/account/data/model/account_model.dart';
import 'package:paisa/features/category/data/data_sources/category_data_source.dart';
import 'package:paisa/features/category/data/model/category_model.dart';
import 'package:paisa/features/recurring/domain/repository/recurring_repository.dart';
import 'package:paisa/features/transaction/data/data_sources/local/transaction_data_manager.dart';
import 'package:paisa/features/transaction/data/model/transaction_model.dart';

final getIt = GetIt.instance;

Future<void> main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await configInjector(getIt);
  await _addDummyData();
  getIt<RecurringRepository>().checkForRecurring();
  runApp(const PaisaApp());
  if (TargetPlatform.android == defaultTargetPlatform ||
      TargetPlatform.iOS == defaultTargetPlatform) {
    initAppShortcuts();
  }
}

_addDummyData() async {
  final accountDataSource = getIt<AccountDataSource>();
  final categoryDataSource = getIt<CategoryDataSource>();
  final transactionDataSource = getIt<TransactionDataSource>();

  await accountDataSource.clear();
  await categoryDataSource.clear();
  await transactionDataSource.clear();

  for (int i = 0; i < 10; i++) {
    await accountDataSource.add(
      AccountModel(
        bankName: 'Bank Name $i',
        name: 'Holder name $i',
        cardType: CardType.values[Random().nextInt(3)],
        color: Colors.amber.value,
      ),
    );

    await categoryDataSource.add(
      CategoryModel(
        name: 'Category name $i',
        color: Colors.amber.value,
        icon: Icons.car_crash.codePoint,
      ),
    );
  }
  final random = Random();
  final startDate = DateTime(2010);
  final endDate = DateTime.now();

  for (int i = 0; i < 100; i++) {
    int accountId = Random().nextInt(10);
    int categoryId = Random().nextInt(10);
    final difference = endDate.difference(startDate).inDays;
    final randomDay = random.nextInt(difference);
    final randomDate = startDate.add(Duration(days: randomDay));
    await transactionDataSource.add(TransactionModel(
      name: 'Transaction name $i',
      time: randomDate,
      accountId: accountId,
      categoryId: categoryId,
      currency: Random().nextDouble() * 100000,
      type: TransactionType.values[Random().nextInt(2)],
    ));
  }
}
main.dart

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @Madhan790 !

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the [GitHub product Discussions](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/about-discussions). This question should be in the Programming Help category. I've gone ahead and moved it for you. Good luck!
